pkg/config/cfgdispatch.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+type configMap map[string]interface{}
+
+func (cm configMap) GetString(key string) (string, error) {
+	val, ok := cm[key]
+	if !ok {
+		return "", nil
+	}
+	s, ok := val.(string)
+	if !ok {
+		return "", fmt.Errorf("key %q has non-string value %T", key, val)
+	}
+	return s, nil
+}
+
+func (cm configMap) GetInt(key string) (int, error) {
+	val, ok := cm[key]
+	if !ok {
+		return 0, nil
+	}
+	j, ok := val.(float64)
+	if !ok {
+		return 0, fmt.Errorf("key %q has non-float64 value representation %T", key, val)
+	}
+	return int(j), nil
+}
+
+func (cm configMap) GetBool(key string) (bool, error) {
+	val, ok := cm[key]
+	if !ok {
+		return false, nil
+	}
+	b, ok := val.(bool)
+	if !ok {
+		return false, fmt.Errorf("key %q has non-bool value %T", key, val)
+	}
+	return b, nil
+}
+
+func (cm configMap) GetDuration(key string) (time.Duration, error) {
+	val, ok := cm[key]
+	if !ok {
+		return 0, nil
+	}
+	j, ok := val.(float64)
+	if !ok {
+		return 0, fmt.Errorf("key %q has non-float64 value representation %T", key, val)
+	}
+	return time.Duration(j), nil
+}
+
+func dispatchConfObj(obj map[string]interface{}, pArgs *ProgArgs) error {
```

Comment:
IIUC, the object we expect here is bunch of `yaml` files serialized into dotted format composed into `ProgArgs` object. 
Why can't we just unmarshal the yaml into `ProgArgs`?

Reply:
explained in the newly-added comments